osmoutils/coin_helper.go

```diff
@@ -127,3 +128,21 @@ func ConvertCoinArrayToCoins(coinArray []sdk.Coin) sdk.Coins {
 	}
 	return coins
 }
+
+func ConvertCoinsToCoinArray(coins sdk.Coins) []sdk.Coin {
+	if coins == nil {
+		return []sdk.Coin(nil)
+	}
+
+	coinArray := make([]sdk.Coin, len(coins))
+	for i, coin := range coins {
+		coinArray[i] = coin
+	}
+
+	// Sort the coinArray lexicographically by Denom
+	sort.Slice(coinArray, func(i, j int) bool {
+		return coinArray[i].Denom < coinArray[j].Denom
+	})
```

Comment:
Can we expect `sdk.Coins` to be already sorted?

Reply:
The issue is it returns it lexicographically backwards due to it ranging over the sorted coins first. So I could probably just range backwards instead

Reply:
Ranged backwards and in turn got rid of the sorting logic https://github.com/osmosis-labs/osmosis/pull/7240/commits/e3c0e3382b51e8dd7fba71febd72d76aef8f09d2